compiler/rustc_const_eval/src/const_eval/eval_queries.rs

```diff
@@ -380,7 +380,11 @@ pub fn eval_in_interpreter<'mir, 'tcx>(
         }
         Ok(mplace) => {
             // Since evaluation had no errors, validate the resulting constant.
+
+            // Temporarily allow access to the static_root_ids for the purpose of validation.
+            let static_root_ids = ecx.machine.static_root_ids.take();
             let res = const_validate_mplace(&ecx, &mplace, cid);
+            ecx.machine.static_root_ids = static_root_ids;
```

Comment:
in case anyone stumbles upon this in the future, this was a rebase screwup and is getting removed again in https://github.com/rust-lang/rust/pull/122397#discussion_r1521914662